src/arch/riscv/insts/vector.hh

```diff
@@ -114,24 +118,47 @@ class VectorMacroInst : public RiscvMacroInst
   protected:
     uint32_t vl;
     uint8_t vtype;
+    uint32_t vlen;
     VectorMacroInst(const char* mnem, ExtMachInst _machInst,
-                   OpClass __opClass)
+                    OpClass __opClass)
         : RiscvMacroInst(mnem, _machInst, __opClass),
         vl(_machInst.vl),
         vtype(_machInst.vtype8)
     {
         this->flags[IsVector] = true;
     }
+
+    VectorMacroInst(const char* mnem, ExtMachInst _machInst,
+                   uint32_t _vlen, OpClass __opClass)
+        : RiscvMacroInst(mnem, _machInst, __opClass),
+        vl(_machInst.vl),
+        vtype(_machInst.vtype8),
+        vlen(_vlen)
+    {
+        this->flags[IsVector] = true;
+    }
```

Comment:
This part is OK to me, but I think we can just leave one constructor with default `vlen`.